subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/BaselineVersion.groovy

```diff
@@ -56,19 +59,39 @@ class BaselineVersion implements VersionResults {
             } else {
                 sb.append "Speed $displayName: Results were inconclusive"
             }
-            String confidencePercent = confidenceInDifference(results.totalTime, current.totalTime) * 100 as int
-            sb.append(" with " + confidencePercent + "% confidence.\n")
+            Number regressionPercentage = PrettyCalculator.percentChange(currentVersionMean, thisVersionMean)
+            String confidence = "${confidenceInDifference(results.totalTime, current.totalTime) * 100 as int}%"
+            sb.append(" with $confidence confidence.\n")
 
             def diff = currentVersionMean - thisVersionMean
             def desc = diff > Duration.millis(0) ? "slower" : "faster"
-            sb.append("Difference: ${diff.abs().format()} $desc (${toMillis(diff.abs())}), ${PrettyCalculator.percentChange(currentVersionMean, thisVersionMean)}%\n")
+            sb.append("Difference: ${diff.abs().format()} $desc (${toMillis(diff.abs())}), ${regressionPercentage}%\n")
             sb.append(current.speedStats)
             sb.append(results.speedStats)
             sb.append("\n")
-            sb.toString()
+            sb.append(dataForMachine(current, results, regressionPercentage.doubleValue(), confidence))
```

Comment:
Let's not add this magic data to the test output. Either use the test database or write the data into a different artifact and archive that artifact for the individual performance tests.